admin/commands/execution/stop_at_height.go

```diff
@@ -0,0 +1,82 @@
+package execution
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/admin"
+	"github.com/onflow/flow-go/admin/commands"
+)
+
+var _ commands.AdminCommand = (*StopAtHeightCommand)(nil)
+
+// StopAtHeightCommand will send a signal to engine to stop/crash EN
+// at given height
+type StopAtHeightCommand struct {
+	height *atomic.Uint64
+	crash  *atomic.Bool
+}
+
+func NewStopAtHeightCommand(height *atomic.Uint64, crash *atomic.Bool) *StopAtHeightCommand {
+	return &StopAtHeightCommand{
+		height: height,
+		crash:  crash,
+	}
+}
+
+type StopAtHeightReq struct {
+	height uint64
+	crash  bool
+}
+
+func (s *StopAtHeightCommand) Handler(ctx context.Context, req *admin.CommandRequest) (interface{}, error) {
+	sah := req.ValidatorData.(StopAtHeightReq)
+
+	oldHeight := s.height.Swap(sah.height)
+	oldCrash := s.crash.Swap(sah.crash)
+
+	log.Info().Msgf("admintool: EN will stop at height %d and crash: %t, previous values: %d %t", sah.height, sah.crash, oldHeight, oldCrash)
+
+	return "ok", nil
+}
+
+func (s *StopAtHeightCommand) Validator(req *admin.CommandRequest) error {
+
+	sahr := StopAtHeightReq{}
+
+	input, ok := req.Data.(map[string]interface{})
+	if !ok {
+		return commands.ErrValidatorReqDataFormat
+	}
+	result, ok := input["height"]
+	errInvalidHeightValue := fmt.Errorf("invalid value for \"height\": expected a positive integer, but got: %v", result)
+
+	if !ok {
+		return errInvalidHeightValue
+	}
+
+	height, ok := result.(int)
+	if !ok || height <= 0 {
+		return errInvalidHeightValue
+	}
+	sahr.height = uint64(height)
```

Comment:
nit: using generics may shorten this block, but no strong preference here.

Reply:
Hmm, can you show an example? We parse unknown object here and want to handle types mismatch, so I'm not sure how generics can help, but I don't have much experience with them

Reply:
sth in my mind like this:

```
func CheckAndGetValue[T any](input map[string]interface{}, key string) (*T, error) {
	result, ok := input[key]
	if !ok {
		return nil, errors.New("no such key in map")
	}

	retVal, ok := result.(T)
	if !ok {
		fmt.Println("not able to convert")
		return nil, errors.New("test")
	}

	return &retVal, nil
}
```

just to exchange some thoughts, not blocking for me

Reply:
Hmm, I experimented a bit with that, but it didn't increase clarity that much (for example we do check if height > 0, which mean with need either extra check anyway - and constructing another error string.
We should either use some proper marshalling/validating framework for mapping whole objects, but that's far outside the scope.
But thanks for the suggestion!